app/models/document.rb

```diff
@@ -256,6 +256,14 @@ def withdraw
     end
   end
 
+  def discard
+    handle_remote_error { discard! }
+  end
+
+  def discard!
+    Services.publishing_api.discard_draft(content_id, previous_version: previous_version)
+  end
```

Comment:
This pair of methods has different semantics to the other publishing-api write methods.  `save`, `publish!`, and `unpublish` all use `handle_remote_error` to wrap the API calls and return `true` or `false` rather than raise.  It's not clear why `publish!` has a bang when the others don't.  

I'm not sure we're following the `ActiveRecord` pattern of `!` meaning a method that might raise and the unadorned method meaning one that won't.  It's a familiar pattern though so maybe we should?  On the other hand do we even need the raise variants, we're not currently using them and it's not clear we'd want to?

Either way, this new pair of methods stands out as odd compared to what we have and we should choose how to make discarding feel more similar to the other write methods.